test/fuzzyDateTime.js

```diff
@@ -0,0 +1,205 @@
+	[() => past.minutes(23), '23 min. ago', () => 1 * minute, { rtf: enAlwaysShort }],
+];
+
+describe('fuzzyDateTime', () => {
+
+	describe('formatFuzzyDate', () => {
+		it('omits the time when formatting an absolute date', () => {
+			expect(formatFuzzyDate(past.days(5), { origin: noon })).to.match(shortDateRegex);
+			expect(formatFuzzyDate(future.days(5), { origin: noon })).to.match(shortDateRegex);
+		});
+
+		it('omits the time when formatting a long absolute date', () => {
+			expect(formatFuzzyDate(past.days(5), { origin: noon, absoluteFormat: 'full' })).to.match(fullDateRegex);
+			expect(formatFuzzyDate(future.days(5), { origin: noon, absoluteFormat: 'full' })).to.match(fullDateRegex);
+		});
+	});
+
+	describe('formatFuzzyDateTime', () => {
+		it('includes the time when formatting an absolute date', () => {
+			expect(formatFuzzyDateTime(past.days(5), { origin: noon })).to.match(shortDateTimeRegex);
+			expect(formatFuzzyDateTime(future.days(5), { origin: noon })).to.match(shortDateTimeRegex);
+		});
+
+		it('includes the time when formatting a long absolute date', () => {
+			expect(formatFuzzyDateTime(past.days(5), { origin: noon, absoluteFormat: 'full' })).to.match(fullDateTimeRegex);
+			expect(formatFuzzyDateTime(future.days(5), { origin: noon, absoluteFormat: 'full' })).to.match(fullDateTimeRegex);
+		});
+
+	});
+
+	describe('formatFuzzy', () => {
+		describe('the readme examples', () => {
+
+			const origin = new Date(2015, 8, 23, 14, 5, 30);
+
+			const ruAlwaysShort = new Intl.RelativeTimeFormat('ru', {
+				localeMatcher: 'best fit',
+				numeric: 'always',
+				style: 'short'
+			});
+
+			it('example 1', () => {
+				expect(formatFuzzyDateTime(
+					new Date(2015, 8, 23, 14, 5, 18),
+					{ origin }
+				)).to.equal('12 seconds ago');
+			});
+
+			it('example 2', () => {
+				expect(formatFuzzyDateTime(
+					new Date(2015, 8, 24, 14, 5, 30),
+					{ origin, rtf: ruAlwaysShort }
+				)).to.equal('через 1 дн.');
+			});
+
+			it('example 3', async() => {
+				const invocations = [];
+				const actualDeltas = [];
+				const onUpdate = (text, opt) => {
+					invocations.push(text);
+					actualDeltas.push(opt.nextUpdateInMilliseconds);
+					return false;
+				};
+				const d = new Date(2015, 8, 23, 14, 5, 25);
+				const deltas = [
+					0,
+					25 * second,
+					1 * minute,
+					1 * minute,
+				];
+
+				let totalDelta = 0;
+				for (const delta of deltas) {
+					totalDelta += delta;
+					formatFuzzyDate(d, {
+						origin: new Date(origin.getTime() + totalDelta),
+						onUpdate
+					});
+				}
+				formatFuzzyDate(d, {
+					origin: new Date(origin.getTime() + 4 * day),
+					onUpdate,
+				});
+
+				deltas.push(1 * minute);
+				deltas.push(-1);
+				expect(actualDeltas).to.deep.equal(deltas.slice(1));
+				expect(invocations).to.deep.equal([
+					'5 seconds ago',
+					'this minute',
+					'1 minute ago',
+					'2 minutes ago',
+					'9/23/2015',
```

Comment:
Removed the other two functions which tested onUpdate and relied on lengthy timeouts timeouts. I tried mocking the setTimeout function, but the results were wrong because the tests relied on the value of `new Date()` changing. Mocking that as well got complicated, and I decided the tests wouldn't add much value.

This test, I think sufficiently tests the timeouts without introducing delays